extensions/wikia/DefaultMessages/DefaultMessagesCache.php

```diff
@@ -0,0 +1,521 @@
+		fclose( $file );
+		@chmod( $filename, 0666 );
+	}
+
+	function saveToScript( $array, $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$filename = $this->mLocalMessageCache . "/messages-$wgDefaultMessagesDB-$code";
+		$tempFilename = $filename . '.tmp';
+		wfMkdirParents( $this->mLocalMessageCache, 0777 ); // might fail
+
+		wfSuppressWarnings();
+		$file = fopen( $tempFilename, 'w');
+		wfRestoreWarnings();
+
+		if ( !$file ) {
+			wfDebug( "Unable to open local cache file for writing\n" );
+			return;
+		}
+
+		fwrite($file,"<?php\n//$hash\n\n \$this->mCache = array(");
+
+		foreach ($array as $key => $message) {
+			$key = $this->escapeForScript($key);
+			$messages = $this->escapeForScript($message);
+			fwrite($file, "'$key' => '$message',\n");
+		}
+
+		fwrite($file,");\n?>");
+		fclose($file);
+		rename($tempFilename, $filename);
+	}
+
+	function escapeForScript($string) {
+		$string = str_replace( '\\', '\\\\', $string );
+		$string = str_replace( '\'', '\\\'', $string );
+		return $string;
+	}
+
+	/**
+	 * Set the cache to $cache, if it is valid. Otherwise set the cache to false.
+	 */
+	function setCache( $cache, $code ) {
+		if ( isset( $cache['VERSION'] ) && $cache['VERSION'] == MSG_CACHE_VERSION ) {
+			$this->mCache[$code] = $cache;
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Loads messages from caches or from database in this order:
+	 * (1) local message cache (if $wgLocalMessageCache is enabled)
+	 * (2) memcached
+	 * (3) from the database.
+	 *
+	 * When succesfully loading from (2) or (3), all higher level caches are
+	 * updated for the newest version.
+	 *
+	 * Nothing is loaded if  member variable mDisabled is true, either manually
+	 * set by calling code or if message loading fails (is this possible?).
+	 *
+	 * Returns true if cache is already populated or it was succesfully populated,
+	 * or false if populating empty cache fails. Also returns true if MessageCache
+	 * is disabled.
+	 *
+	 * @param $code String: language to which load messages
+	 */
+	function load( $code = false, $useDB = true ) {
+		if ( !$this->mUseCache ) {
+			return true;
+		}
+
+		if( !is_string( $code ) ) {
+			# This isn't really nice, so at least make a note about it and try to
+			# fall back
+			wfDebug( __METHOD__ . " called without providing a language code\n" );
+			$code = 'en';
+		}
+
+		# Don't do double loading...
+		if ( isset($this->mLoadedLanguages[$code]) ) return true;
+
+		# 8 lines of code just to say (once) that message cache is disabled
+		if ( $this->mDisable ) {
+			static $shownDisabled = false;
+			if ( !$shownDisabled ) {
+				wfDebug( __METHOD__ . ": disabled\n" );
+				$shownDisabled = true;
+			}
+			return true;
+		}
+
+		# Loading code starts
+		wfProfileIn( __METHOD__ );
+		$success = false; # Keep track of success
+		$where = array(); # Debug info, delayed to avoid spamming debug log too much
+		$cacheKey = $this->memcKey( 'messages', $code ); # Key in memc for messages
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "messages" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)